File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = [0, 0]
    while s != "1":
        one_count = 0
        for l in s:
            if l == "1":
                one_count += 1
        answer[1] += len(s) - one_count
        s = radix_change(one_count, 2)
        answer[0] += 1
    
    return answer

def radix_change(num, radix):
    if num == 0: return '0'
    nums = []
    while num:
        num, digit = divmod(num, radix)
        nums.append(str(digit))
    return "".join(reversed(nums))

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = {0, 0};
        
        while(!s.equals("1")){
            int oneCount = count(s);
            answer[0] += 1;
            answer[1] += s.length() - oneCount;
            s = Integer.toString(oneCount, 2);
        }
        
        return answer;
    }
    
    private int count(String s){
        int count = 0;
        char[] arr = s.toCharArray();
        for(char c : arr){
            if(c == '1'){
                count += 1;
            }
        }
        return count;
    }
}